code/modules/mapping/writer.dm

```diff
@@ -0,0 +1,178 @@
+/**Map exporter
+* Inputting a list of turfs into convert_map_to_tgm() will output a string
+* with the turfs and their objects / areas on said turf into the TGM mapping format
+* for .dmm files. This file can then be opened in the map editor or imported
+* back into the game.
+* ============================
+* This has been made semi-modular so you should be able to use these functions
+* elsewhere in code if you ever need to get a file in the .dmm format
+**/
+/atom/proc/get_save_vars()
+	return list("pixel_x", "pixel_y", "dir", "name", "req_access", "req_access_txt", "piping_layer", "color", "icon", "icon_state", "pipe_color", "amount")
+
+GLOBAL_LIST_INIT(save_file_chars, list(
```

Comment:
Couldn't this be a regex instead? `[a-zA-Z]`

Reply:
Don't think so, these aren't used to find instances that these are found, they're used to find the next char to use when generating a TGM key.

Reply:
Ah, that makes sense. LGTM